pkg/client/client.go

```diff
@@ -353,12 +382,21 @@ func (c *client) Get(ctx context.Context, key ObjectKey, obj Object, opts ...Get
 	}
 }
 
+// shouldReturnCacheErr determines if we should return the error we got from the cache, which we do in every case
+// except for when we're configured to do live lookups and the cache returns cache.ErrResourceNotCached
+func (c *client) shouldReturnCacheErr(err error) bool {
+	var errNotCached *cacheerrors.ErrResourceNotCached
+	return !(c.cacheMissPolicy == UncachedResourcePolicyLiveLookup && errors.As(err, &errNotCached))
```

Comment:
This hurts my brain 😄.

Reply:
I had it the other way around but that broke Vince's brain, not sure if there's a good middle ground :|